openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -407,10 +405,11 @@ class CoursewareInformation(RetrieveAPIView):
             * masquerading_expired_course: (bool) Whether this course is expired for the masqueraded user
             * upgrade_deadline: (str) Last chance to upgrade, in ISO 8601 notation (or None if can't upgrade anymore)
             * upgrade_url: (str) Upgrade linke (or None if can't upgrade anymore)
-        course_goals:
-            selected_goal:
-                days_per_week: (int) The number of days the learner wants to learn per week
-                subscribed_to_reminders: (bool) Whether the learner wants email reminders about their goal
+        * course_goals:
+            * selected_goal:
+                * days_per_week: (int) The number of days the learner wants to learn per week
+                * subscribed_to_reminders: (bool) Whether the learner wants email reminders about their goal
+            * weekly_learning_goal_enabled: Flag indicating if this feature is enabled for this call
```

Comment:
I kept the flag in the api, and intentionally  didn't change the name to match the waffle flag. It was going to complicate deployment. Is this controversial?

Reply:
I don't mind having a flag around with a slightly old but still sensible name, if it makes life easier.

It does feel a little awkward the way things will work now though: in this view, `course_goals` is either null or a dict with this flag set to True. In the outline view, `course_goals` is always a dict, where this flag is either missing or True (which feels like an odd dichotomy). (Did I get that right?)

Feels like they could/should use similar structure? Would that complicate deployment to at least unify / add some sense to that?

And both views could probably stand to just be calling a single method to get this data struct, rather than re-implementing it twice, but subtly differently.

Reply:
I think you are right, that this could use some unification. I am planning on tackling it as the first part of the API unification, since I think one of the APIs will lose this data. No need to worry about consistency if there is only one!